posts/2022-04-15-libertydevops-gitjenkinshelmiks.adoc

```diff
@@ -0,0 +1,192 @@
+---
+layout: post
+title: "Continuous integration and delivery (CI/CD) of cloud-native Java applications using Jenkins"
+categories: blog
+author_picture: https://avatars3.githubusercontent.com/shamjithantholi
+author_github: https://github.com/shamjithantholi
+seo-title: Continuous integration and delivery (CI/CD) of cloud-native Java applications using Jenkins - OpenLiberty.io
+seo-description: How to build a DevOps pipeline for a cloud-native Java application with Jenkins as the core controller component.
+blog_description: How to build a DevOps pipeline for a cloud-native Java application with Jenkins as the core controller component.
+open-graph-image: https://openliberty.io/img/blog/liberty-devops-generic-architecture-2.png
+---
+= Continuous integration and delivery (CI/CD) of cloud-native Java applications using Jenkins
+Shamjith Antholi <https://github.com/shamjithantholi>
+:imagesdir: /
+:url-prefix:
+:url-about: /
+
+[#Intro]
+
+Continuous integration and continuous deployment (CI/CD) is a process in which developers frequently merge code changes into a source code repository, which then automatically builds the code, runs security scans, runs tests, and then deploys the application. Automation of this process, through adopting a DevOps culture, tools, and processes, enables organizations to release on a more frequent basis without compromising on quality. There are various tools available to automate the CI/CD process that can be readily integrated into a DevOps deployment pipeline. In this blog post we'll discuss how you can use Jenkins to automate the CI/CD process for building, testing, and deploying your cloud-native Java applications running on link:https://openliberty.io[Open Liberty].
+
+Jenkins is a popular open source automation server that has hundreds of plugins to help integrate it into your existing DevOps pipeline. The following diagram shows a simple architecture for a Jenkins deployment pipeline that builds and deploys a Jakarta EE and MicroProfile application in a single Kubernetes environment. When a code change in the Git repository is detected by Jenkins, Jenkins builds the code, runs Junit unit tests and then SonarQube security and static analysis scans against the built code. Jenkins then generates a Docker image of the application with its Open Liberty server, scans the image with Trivy, and saves the image to a Docker repository. Jenkins then triggers a deployment using CLI commands and deploys the Docker image to Kubernetes.
+
```

Comment:
Should we consider a reference to GitOps like the below?

"Since Git is our starting point and the CI/CD process is triggered by source changes in Git, this is also known as the GitOps approach to DevOps."